optimade/server/models/jsonapi.py

```diff
@@ -1,21 +1,85 @@
 """
 This module should reproduce https://jsonapi.org/schema
 """
-from typing import Optional
-
-from pydantic import BaseModel, UrlStr, Union
 
+from typing import Optional, Set, Union, Dict, Any
+from pydantic import BaseModel, UrlStr, constr
 
 class Link(BaseModel):
     href: UrlStr
     meta: Optional[dict]
 
-
 class Links(BaseModel):
     next: Optional[Union[UrlStr,Link]]
     base_url: Optional[Union[UrlStr,Link]]
```

Comment:
would you agree that this should be moved into an "optimade" links class?
I guess base_url is not part of the jsonapi links spec?

Reply:
Yes I'd move that into an "optimade". The json representation of Links is more general
"links": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/definitions/link"
      }
    },